libconcurrent/primitives/threadtools.c

```diff
@@ -59,33 +75,79 @@ inline static void *kthreadWrapper(void *arg) {
     return NULL;
 }
 
-inline uint32_t synchPreferedCoreOfThread(uint32_t pid) {
-    uint32_t prefered_core = 0;
+inline uint32_t synchPreferredCoreOfThread(uint32_t pid) {
+    uint32_t preferred_core = 0;
+
+    if (__schedule_policy == SYNCH_THREAD_PLACEMENT_FLAT) {
+        preferred_core = pid;
+    } else {
 #ifdef SYNCH_NUMA_SUPPORT
-    int ncpus = numa_num_configured_cpus();
-    int nodes = numa_num_task_nodes();
-    int node_size = ncpus / nodes;
-
-    if (numa_node_of_cpu(0) == numa_node_of_cpu(ncpus / 2)) {
-        int half_node_size = node_size / 2;
-        int offset = 0;
-        uint32_t half_cpu_id = pid;
-
-        if (pid >= ncpus / 2) {
-            half_cpu_id = pid - ncpus / 2;
-            offset = ncpus / 2;
+        uint32_t ncpus = numa_num_configured_cpus();
+        uint32_t nodes = numa_num_task_nodes();
```

Comment:
Is this the number of numa nodes?
What's the difference between task nodes and numa nodes?

Reply:
numa_num_task_nodes returns the actual number of nodes in the system.
The numa_nodes field in the structures above is that a user may want to enforce the system to behave with a different amount of NUMA nodes. 
For example, the system may be equipped with 8 NUMA nodes, but the user may want to create a fictitious topology with just 4 NUMA nodes. In some cases this may improve the performance, e.g., multi-level topologies etc.

Reply:
then consider renaming:
`numa_nodes` --> `configured_numa_nodes` or `accessible_numa_nodes`
`numa_num_task_nodes()` --> `numa_num_physical_nodes()`

Reply:
It's actually the API of `libnuma`. So, no change to change it.